includes/wikia/models/DesignSystemGlobalNavigationModel.class.php

```diff
@@ -0,0 +1,286 @@
+		}
+
+		return $data;
+	}
+
+	private function getHref( $hrefKey ) {
+		return DesignSystemSharedLinks::getHref( $hrefKey, $this->lang );
+	}
+
+	private function getAnonUserData() {
+		return [
+			'header' => [
+				'type' => 'line-image',
+				'image' => 'wds-icons-user',
+				'title' => [
+					'type' => 'translatable-text',
+					'key' => 'global-navigation-userinfo-anon-avatar-title',
+				],
+			],
+			'links' => [
+				[
+					'type' => 'link-authentication',
+					'title' => [
+						'type' => 'translatable-text',
+						'key' => 'global-navigation-userinfo-signin-title',
+					],
+					'href' => $this->getHref( 'user-signin' ),
+					'param-name' => 'redirect',
+				],
+				[
+					'type' => 'link-authentication',
+					'title' => [
+						'type' => 'translatable-text',
+						'key' => 'global-navigation-userinfo-register-title',
+					],
+					'subtitle' => [
+						'type' => 'translatable-text',
+						'key' => 'global-navigation-userinfo-register-description',
+					],
+					'href' => $this->getHref( 'user-register' ),
+					"param-name" => "redirect"
+				],
+			],
+		];
+	}
+
+	private function getLoggedInUserData( $user ) {
+		global $wgEnableWallExt;
+
+		$userName = $user->getName();
+
+		return [
+			'header' => [
+				'type' => 'avatar',
+				'username' => [
+					'type' => 'text',
+					'value' => $userName
+				],
+				'url' => AvatarService::getAvatarUrl( $userName, 50 ),
+			],
+			'links' => [
+				[
+					'type' => 'link-text',
+					'href' => $this->getPageUrl( $userName, NS_USER ),
+					'title' => [
+						'type' => 'translatable-text',
+						'key' => 'global-navigation-userinfo-profile'
+					]
+				],
+				[
+					'type' => 'link-text',
+					'href' => $wgEnableWallExt
+						? $this->getPageUrl( $userName, NS_USER_WALL )
+						: $this->getPageUrl( $userName, NS_USER_TALK ),
+					'title' => [
+						'type' => 'translatable-text',
+						'key' => 'global-navigation-userinfo-talk'
+					]
+				],
+				[
+					'type' => 'link-text',
+					'href' => $this->getPageUrl( 'Preferences', NS_SPECIAL ),
+					'title' => [
+						'type' => 'translatable-text',
+						'key' => 'global-navigation-userinfo-preferences'
+					]
+				],
+				[
+					'type' => 'link-text',
+					'href' => $this->getPageUrl( 'Contents', NS_HELP ),
+					'title' => [
+						'type' => 'translatable-text',
+						'key' => 'global-navigation-userinfo-help'
+					]
+				],
+				[
+					'type' => 'link-authentication',
+					'href' => $this->getPageUrl( 'UserLogout', NS_SPECIAL ),
+					'title' => [
+						'type' => 'translatable-text',
+						'key' => 'global-navigation-userinfo-signout'
+					],
+					'param-name' => 'returnto'
+				]
+			],
+		];
+	}
+
+	private function getNotifications( $user ) {
+		global $wgEnableWallExt;
+
+		$userName = $user->getName();
+
+		return [
+			'module' => [
+				'type' => 'notifications',
+				'url' => $wgEnableWallExt
+					? $this->getPageUrl( $userName, NS_USER_WALL )
```

Comment:
Is that the correct URL? The Api URL for fetching notifications is not a MW page...
http://muppet.wikia.com/wikia.php?controller=WallNotificationsExternal&method=getUpdateCounts&format=json
http://muppet.wikia.com/wikia.php?controller=WallNotificationsExternal&method=getUpdateWiki&format=json&isCrossWiki=0&wikiId=831